openapi/v2/core-command.yaml

```diff
@@ -71,47 +71,134 @@ components:
+    SimpleReading:
+      description: "An event reading for a simple data type"
       allOf:
-        - $ref: '#/components/schemas/BaseRequest'
-      description: "A request type specifying an individual command for execution."
-      type: object
+        - $ref: '#/components/schemas/BaseReading'
+        - type: object
+          properties:
+            value:
+              description: "A string representation of the reading's value"
+              type: string
+      required:
+        - value
+    BinaryReading:
+      description: "An event reading for a binary data type"
+      allOf:
+        - $ref: '#/components/schemas/BaseReading'
+        - type: object
+          properties:
+            binaryValue:
+              description: "If the value of the reading is binary, it will be found in this property as a byte array"
+              type: string
+              format: byte
+            mediaType:
+              description: "E.g. MIME Type, indicates what the content type of the binaryValue property is if it's populated."
+              type: string
+          required:
+            - binaryValue
+            - mediaType
+    Event:
+      description: "A discrete event containing one or more readings"
       properties:
-        deviceId:
+        apiVersion:
+          description: "A version number shows the API version in DTOs."
           type: string
-          format: uuid
-          description: "Refers to the device targeted by the command. Could be either the device ID or name."
-        commandId:
+        id:
+          description: "The unique identifier for the event"
           type: string
           format: uuid
-          description: "Refers to the command being invoked. Could be either the command ID or name."
+        deviceName:
+          description: "The name of the device from which the event originated"
+          type: string
+        profileName:
+          description: "The name of the device profile from which the event originated"
+          type: string
+        created:
+          description: "A Unix timestamp indicating when (if) the event was initially persisted to a database."
+          type: integer
+        origin:
+          description: "A Unix timestamp indicating when the event was originated at the source device (can support nanoseconds)"
+          type: integer
+        readings:
+          description: "One or more readings captured at the time of the event"
+          type: array
+          items:
+            $ref: '#/components/schemas/SimpleReading'
+        tags:
+          description: "List of zero or more Tags attached to the Event which give more context to the Event"
+          title: tags
+          type: object
+          example: {
+            "Gateway-id": "HoustonStore-000123",
+            "Latitude": "29.630771",
+            "Longitude": "-95.377603",
+          }
       required:
-      - device
-      - command
-    IssueCommandResponse:
+        - id
```

Comment:
Please add apiVersion to list of required.

Reply:
Added.